app/models/post.rb

```diff
@@ -0,0 +1,16 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: :author_id
+  has_many :comments, foreign_key: :post_id
+  has_many :likes, foreign_key: :post_id
+
+  after_save :update_posts_counter_for_user
+
+  def update_posts_counter_for_user
+    counter = Post.count('author_id')
+    author.update(posts_counter: counter)
+  end
```

Comment:
- [ ] Since we are using the Rails active Callbacks `after_save` it would be better if we make the methods called by the Rails callbacks private, so that only the model will have access to those methods. For example;
```
private
def update_posts_counter_for_user
    counter = Post.count('author_id')
    author.update(posts_counter: counter)
  end
```
Let us kindly implement this for all our after_save methods
![Screenshot from 2023-01-26 14-14-28](https://user-images.githubusercontent.com/46388113/214856412-62604252-1981-4a57-816c-cb6ee594e3a1.png)

Reply:
Thanks @addod19 for your polite constructive feedback.